### ISSUE TITLE
Submitting jobs

### ISSUE THREAD

**Issue by woodoo46** [marbl/canu#861]
Hi Canu Developers,

I found a weird behaviour, or I have used the command wrong. At the trimming step, the program estimates that it needs to submit 3674 tasks, so that the program submits an array job id from 1-3674. Due to the limitation of our queue, only the first 3000 tasks finished (array id from 1-3000). When I reran the command, the program always submits an array job with array id from 1-674, but it should be 3001-3674?

Thanks.

George

**Comment by skoren** [comment]
Canu jobs support an offset passed as a command-line argument. So as long as it is including 3000 as the offset, the job array tasks are correct. You can see the submitted command in the relevant log file Canu prints before finishing.

**Comment by woodoo46** [comment]
The following is resubmission script:

qsub -j oe -d `pwd` \
  -l mem=6g -l nodes=1:ppn=2 -l walltime=96:00:00 -l vmem=64gb -l mem=64gb -o overlap.\$PBS_ARRAYID.out \
  -N "obtovl_sample" \
  -t 1-674 \
  ./overlap.sh  \
> ./overlap.jobSubmit-01.out 2>&1

The array id is from 1 to 674, the script always check the result of 1-674, which have been finished, therefore, the next 3001-3674 never had a chance to rerun. I manually added the 3000 on the -t parameters, and the pipeline moved forward.

Any thought?

**Comment by skoren** [comment]
Is this PBS?

**Comment by woodoo46** [comment]
Yes.

**Comment by brianwalenz** [comment]
For the record, this can be fixed by
- adding 3000 to the values in the -t ranges, so -t 3001-3674
- adding 3000 after overlap.sh, so the second to last line becomes `./overlap.sh 3000 \` (if you look carefully, there are two spaces before the backslash; canu failed to put 3000 there)
- running overlap.sh by hand, 'overlap.sh 3000', 'overlap.sh 3001', etc., then restarting canu when all jobs are complete.

Reopen or make a new issue as needed.

**Comment by skoren** [comment]
Canu should recognize that your job is restricted to 3000 and should have submitted two jobs unless your grid is somehow restricting the array sizes in a non-standard way. Canu will parse the output of 
```
qmgr -c 'p s'|grep max_job_array_size
```
You may be able to manually set gridEngineArrayMaxJobs=3000 if the output above isn't configured correctly.

I tested the latest code on PBS  6.1.2 and cannot reproduce this bug. I manually limited the max array size to 10 and submitted 100 jobs. Each job properly had the -F 10, 20, 30, etc offset for each of the 10 array jobs submitted, here is an example:
```
  ./correctReads.sh -F "50" \
```

I also tested killing one of the arrays and letting Canu re-submit it. It correctly submitted it with the offset again:
```
qsub -j oe -d `pwd` \
  -l mem=3g -l nodes=1:ppn=4 -l walltime=4:00:00 \
  -N "cor_asm" \
  -t 1-10 \
  ./correctReads.sh -F "80" \
> ./correctReads.jobSubmit-01.out 2>&1
```

**Comment by woodoo46** [comment]
It might be the PBS version, we are using an older version PBS: **6.0.2.h2**. And also I was using **Canu 1.6**. What I noticed is that the program initially submitted two batches of jobs 1-3000; and another one is 1-674. And then when I tried to rerun the program it repeatedly using array id 1-674, until I manually changed the array id to 3001-3674.  Anyway, I have asked our admin to lift the array size to 4000 for me. 

Thanks.
